backend_plugin/src/main/evaluators/AddImportEvaluator.java

```diff
@@ -0,0 +1,104 @@
+package main.evaluators;
+
+import java.util.Iterator;
+
+import org.eclipse.jface.text.BadLocationException;
+import org.eclipse.jface.text.DocumentEvent;
+import org.eclipse.jface.text.source.Annotation;
+import org.eclipse.jface.text.source.IAnnotationModel;
+
+public class AddImportEvaluator extends FeatureEvaluator {
+	
+	private boolean unresolvedVariablesExist;
+	private DocumentEvent lastDocumentEvent;
+	
+	/**
+	 * Construct an AddImportEvaluator
+	 */
+	public AddImportEvaluator() {
+		this.featureID = "addImportStatementsSuggestion";
+		this.lastDocumentEvent = null;
+		this.unresolvedVariablesExist = false;
+	}
+
+	/**
+	 * Tracks the user's changes, and determines whether they are fixing an
+	 * "unresolved type" issue by manually typing the import statement
+	 * @return boolean true if the user is manually adding an import statement
+	 * 					when unresolved variables exist; false otherwise
+	 */
+	@Override
+	public boolean evaluateDocumentChanges(DocumentEvent docEvent) {
+		
+		if (this.lastDocumentEvent == null) {
+			
+			// This is the first DocumentEvent captured by the Evaluator
+			this.lastDocumentEvent = docEvent;
+		} else if (this.unresolvedVariablesExist) {
+			
+			// Check if the last two DocumentEvents represent the user typing "t"
+			// followed by " "
+			String textAdded = this.lastDocumentEvent.getText() + docEvent.getText();
+			int offsetDifference = docEvent.getOffset() - this.lastDocumentEvent.getOffset();
+			this.lastDocumentEvent = docEvent;
+			if (textAdded.equals("t ") && offsetDifference == 1) {
+				
+				// If the "t" is the end of an import statement, trigger the feature evaluation
+				return this.checkForTypedImportStatement(docEvent);
+			}
+		}
+		
+		// The user did not manually type an import statement
+		return false;
+	}
+	
+	/**
+	 * Checks the previous 6 characters before the offset of the given DocumentEvent
+	 * @param docEvent the DocumentEvent representing the last text insertion
+	 * @return boolean true if the previous 6 characters in the document are
+	 * 					'i'-'m'-'p'-'o'-'r'-'t'; false otherwise
+	 */
+	private boolean checkForTypedImportStatement(DocumentEvent docEvent) {
+		try {
+			
+			// Check if the last six characters in the document spell "import"
+			int startOffset = docEvent.getOffset() - 6;
+			String testing = docEvent.getDocument().get(startOffset, 6);
+			return testing.equals("import");
```

Comment:
We should also check to make sure it's at the start of the document line. This looks like it'll catch "import " written anywhere

Reply:
Updated the code to check that "import" is at the start of the line